flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -27,4 +27,9 @@ public class AvroToProtoSerializer implements BigQueryProtoSerializer<GenericRec
     public ByteString serialize(GenericRecord record) throws BigQuerySerializationException {
         throw new UnsupportedOperationException("serialize method is not supported");
     }
+
+    @Override
+    public void init(BigQuerySchemaProvider schemaProvider) {
+        throw new UnsupportedOperationException("init method is not supported");
```

Comment:
do callers not need to invoke init? ie, is the constructor sufficient? if so, we should include that in the error message.

Reply:
Implementation for `init` has been added in #104 ([here](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/b0ae6c26143991f2b7ac2c747d42e11f7599950e/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java#L54)).

`do callers not need to invoke init?`
We dont want to enforce that. The `init` is an opinionated action that our AvroToProtoSerializer needs to perform based on the BigQuery table's schema. Other serializers (which can be defined by users) may not need this "pre-process" and hence it can be a no-op. Also, we do not want to tie the `BigQuerySchemaProvider` with the serializer's constructor for similar reasons. The schema provider is used for multiple purposes by the sink and we are trying to hide most of it from the user. Hence, the serializer and schema provider are separate members in the sink config.

`is the constructor sufficient?`
If the respective serializer's design is sufficient with just the constructor and no init, then we want to allow that, basically making the `init` a no-op.

Reply:
That being said, would you suggest making the parent `BigQueryProtoSerializer` interface an abstract class instead where the `init` method is empty instead of abstract?

Reply:
Made the above change of defining `BigQueryProtoSerializer` as abstract class instead of interface and making `init` method a no-op unless over-ridden.

Reply:
nice i like that better; thanks! you can resolve this convo